aFWall/src/main/java/dev/ukanth/ufirewall/util/LogNetUtil.java

```diff
@@ -0,0 +1,217 @@
+            super.onPreExecute();
+
+            progress = new MaterialDialog.Builder(context)
+                    .title(R.string.searching)
+                    .content(R.string.looking_for_data)
+                    .progress(true, 0)
+                    .progressIndeterminateStyle(true)
+                    .show();
+        }
+
+        @Override
+        protected String doInBackground(NetParam... params)
+        {
+            start_time = System.currentTimeMillis();
+
+            try{
+
+                switch(params[0].type){
+
+                    case PING:
+
+                        // Ping
+
+                        try{
+
+                            /* [WIP] looking for ping timeout option */
+                            String result = output(Runtime.getRuntime().exec(String.format(PING_CMD, /*G.logPingTimeout(), */params[0].address)));
+
+                            if(result.isEmpty()){
+
+                                result = context.getString(R.string.network_connection_not_available);
+                            }
+
+                            return result;
+
+                        }catch (Exception eex){
+
+                            try{
+
+                                if(InetAddress.getByAddress(params[0].address.getBytes()).isReachable(G.logPingTimeout())){
+
+                                    return String.format(context.getString(R.string.reachable_timeout), finish_time());
+
+                                }
+
+                            }catch(UnknownHostException ex){
+
+                                return String.format("Currently IP(%s) is not Reachable, timeout: %d ms", params[0].address, finish_time());
+                            }
+                        }
+
+                    case RESOLVE:
+
+                        // Resolve
+
+                        try{
+
+                            return InetAddress.getByName(params[0].address).getCanonicalHostName();
+
+                        }catch(UnknownHostException ex){
+
+                            return String.format("Currently can not resolve Host for IP(%s), timeout: %d ms", params[0].address, finish_time());
+                        }
+                }
+
+            }catch (Exception e){
+
+                e.printStackTrace();
+            }
+
+            return context.getString(R.string.error_or_unknown_category);
+        }
+
+        @Override
+        protected void onPostExecute(String s) {
+            super.onPostExecute(s);
+
+            if(onFinishRequest_callback != null){
+                onFinishRequest_callback.then(s);
+            }
+
+            try {
+                if ((progress != null) && progress.isShowing()) {
+                    progress.dismiss();
+                }
+            } catch (IllegalArgumentException e) {
+                e.printStackTrace();
+                // Handle or log or ignore
+            } catch (final Exception e) {
+                e.printStackTrace();
```

Comment:
FYI http://stackoverflow.com/questions/2072148/should-i-remove-e-printstacktrace-from-my-code-before-publishing